packages/pages/src/vite-plugin/build/closeBundle/functionMetadata.ts

```diff
@@ -61,21 +60,40 @@ const getFunctionMetadataMap = async (
 async function generateFunctionMetadata(
   filepath: string
 ): Promise<[string, FunctionMetadata]> {
-  const buildResult = await esbuild.build({
-    entryPoints: [filepath],
-    outdir: TEMP_DIR,
-    write: false,
-    format: "esm",
-    bundle: true,
-  });
-  const importedFile = await importFromString(buildResult.outputFiles[0].text);
+  project.addSourceFileAtPath(filepath);
+  const defaultExportDeclaration = project
+    .getSourceFile(filepath)
+    ?.getDefaultExportSymbol()
+    ?.getDeclarations()[0];
```

Comment:
Not in any standard case that I could come up with.  If there is no default export, then it will returned `undefined`. Even if there is a weird case where the user mistakenly has `export default ;`, `getDeclarations` will still return a node.